apps/video/test/marionette/lib/video.js

```diff
@@ -130,14 +134,79 @@ Video.prototype = {
    * @return {Marionette.Element} First element of all thumbnail images.
    */
   get thumbnail() {
-    return this.client.findElement(Video.Selector.thumbnail);
+    return this.client.helper.waitForElement(Video.Selector.thumbnail);
   },
 
   /**
    * @return {Marionette.Element} video player element.
    */
   get player() {
-    return this.client.findElement(Video.Selector.player);
+    return this.client.helper.waitForElement(Video.Selector.player);
+  },
+
+  /**
+   * When a video begins playing, displays the controls.
+   * Function contains logic specific to when a video begins playing;
+   * not to be used to arbitrarily display the controls during the playing
+   * of a video.
+   */
+  waitForControls: function() {
+    var elapsedElem = this.client.findElement(Video.Selector.elapsedTime);
+
+    // Ensure when we tap on the player view the video controls are not
+    // displayed -- they are displayed when the video first plays and then
+    // they are automatically hidden.
+    //
+    // Wait for controls to appear as the video begins to play. Specify
+    // a timeout to account for the controls having already been displayed
+    // and auto-hidden.
+    this.client.waitFor(function()
+      { return elapsedElem.displayed(); },
+      { interval: 10, timeout: 5000 }
+    );
+
+    // Either the controls are now displayed or they have already been
+    // automatically hidden. In the case where they have been displayed,
+    // wait for them to be automatically hidden.
+    this.client.waitFor(function() { return !elapsedElem.displayed(); });
+
+    // Tap to display the controls and wait for them to be displayed.
+    this.player.tap();
+    this.client.waitFor(function() { return elapsedElem.displayed(); });
+
+    // Wait for the video to play for at least one second (to allow
+    // for validation checks).
+    var getElapsedTimeSecondsFunc = this.getElapsedTimeSeconds.bind(this);
+    this.client.waitFor(function() {
+      return getElapsedTimeSecondsFunc() > 0;
+    });
```

Comment:
It seems a little odd to me to put this wait here in the waitForControls() function rather than in the test itself.

Can we use arrow functions in marionette tests yet? If so, this seems like a perfect place to do so because then you could just use this.getElapsedTimeSeconds() in the function.